### PR TITLE
COMMERCE-6973 Order type service and API implementation 2/3

### PR DESCRIPTION
**Pull request by alessiorendina**
Based on #1554

### CODE REVIEW

**Comment by Commando18** [comment]
ci:test:sf

**Comment by Commando18** [comment]
ci:test:relevant

**Comment by Commando18** [comment]
ci:test:sf

**Comment by Commando18** [comment]
ci:test:relevant

**Comment by Commando18** [comment]
ci:stop

**Comment by Commando18** [comment]
ci:test:sf

**Comment by Commando18** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/search/CommerceOrderTypeSearchRegistrar.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Alessio Antonio Rendina
+ */
+@Component(enabled = false, immediate = true, service = {})
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/search/CommerceOrderTypeSearchRegistrar.java:34]: I believe in you @Commando18

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/resource/v1_0/OrderTypeChannelResourceImpl.java [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+					commerceOrderType.getCommerceOrderTypeId(), null);
+
+		return Page.of(
+			_toOrderTypeChannels(commerceOrderTypeRels), pagination,
+			totalItems);
+	}
+
+	@NestedField(parentClass = OrderType.class, value = "orderTypeChannels")
+	@Override
+	public Page<OrderTypeChannel> getOrderTypeIdOrderTypeChannelsPage(
```
[the remark this answers is not in the document]
Comment (Commando18) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/resource/v1_0/OrderTypeChannelResourceImpl.java:100]: I'll remove it, thanks

**Pull request closed without merging**